libdrgn/helpers.h

```diff
@@ -43,4 +45,106 @@ struct drgn_error *linux_helper_find_task(struct drgn_object *res,
 					  const struct drgn_object *ns,
 					  uint64_t pid);
 
+/**
+ * Iterator convention:
+ *
+ * For all of the iterators defined below, the convention for each of the
+ * `*_next` functions is that upon returning, `*ret` will point to space
+ * allocated inside of `iter`. The caller is free to do what they wish with
+ * this return value, but should note that it will be overwritten the next time
+ * the `*_next` function is called.
+ */
+
+struct radix_tree_iter_frame {
```

Comment:
Sorry, one more thing: let's rename these types and non-static functions to be prefixed with `linux_helper_`. E.g., `struct linux_helper_radix_tree_iter_frame`, `linux_helper_radix_tree_iter_frame_vector`, `linux_helper_radix_tree_iter_init`. It's way more verbose, but:

* We had at least one case where an overly-generic name caused symbol conflicts when statically linking: b0ae2867d51e0e47462ec1a2a96b03d327d7cf58.
* It's more clear that these are specifically for working with the Linux kernel, not generic data structures.

For static functions and types internal to `linux_kernel_helpers.c`, this doesn't matter.

Reply:
Do you still want the effected code wrapped at 80 columns, or can I bump that up to say 100 columns for this section to help preserve readability?

Reply:
The latter, I'd prefer that you go over the column limit rather than do crazy wrapping things.